tests/core/validation/test_validation.py

```diff
@@ -0,0 +1,391 @@
+        ({}, True),
+        (set(), True),
+        ([1, 2, 3], True),
+        ([1, 2, 1], False),
+        (['a', 'b', 'c'], True),
+        (['1', '2', '3'], True),
+        (['1', '2', '1'], False),
+        (['1', '2', 1], True),
+    ),
+)
+def test_validate_unique(value, is_valid):
+    if is_valid:
+        validate_unique(value)
+    else:
+        with pytest.raises(ValidationError):
+            validate_unique(value)
+
+
+@pytest.mark.parametrize(
+    "value,is_valid",
+    (
+        (tuple(), False),
+        ([], False),
+        ({}, False),
+        (set(), False),
+        (1, False),
+        (True, False),
+        ('abc', False),
+        (b'abc', True),
+    ),
+)
+def test_validate_is_bytes(value, is_valid):
+    if is_valid:
+        validate_is_bytes(value)
+    else:
+        with pytest.raises(ValidationError):
+            validate_is_bytes(value)
+
+
+@pytest.mark.parametrize(
+    "value,is_valid",
+    (
+        (tuple(), False),
+        ([], False),
+        ({}, False),
+        (set(), False),
+        (1234, True),
+        ('abc', False),
+        (True, False),
+    ),
+)
+def test_validate_is_integer(value, is_valid):
+    if is_valid:
+        validate_is_integer(value)
+    else:
+        with pytest.raises(ValidationError):
+            validate_is_integer(value)
+
+
+@pytest.mark.parametrize(
+    "value,is_valid",
+    (
+        (tuple(), False),
+        ([], False),
+        ({}, False),
+        (set(), False),
+        ('a', False),
+        (1, False),
+        (True, True),
+        (False ,True),
+    ),
+)
+def test_validate_is_boolean(value, is_valid):
+    if is_valid:
+        validate_is_boolean(value)
+    else:
+        with pytest.raises(ValidationError):
+            validate_is_boolean(value)
+            
+
+@pytest.mark.parametrize(
+    "value,length,is_valid",
+    (
+        (tuple(), 0, True),
+        ([1], 1, True),
+        ({'A': 'B', 'C': 1}, 3, False),
+        (set(['A', 'B', 1, 2]), 4, True),
+        ('abcde', 4, False),
+        (b'123', 3, True),
+        (range(1,10), 9, True)
+    ),
+)
+def test_validate_length(value, length, is_valid):
+    if is_valid:
+        validate_length(value, length)
+    else:
+        with pytest.raises(ValidationError):
+            validate_length(value, length)
+
+
+@pytest.mark.parametrize(
+    "value,maximum_length,is_valid",
+    (
+        (tuple(), 0, True),
+        ([1], 0, False),
+        ({'A': 'B', 'C': 1}, 3, True),
+        (set(['A', 'B', 1, 2]), 2, False),
+        ('abcde', 5, True),
+        (b'123', 3, True),
+        (range(1,10), 15, True),
+    ),
+)
+def test_validate_length_lte(value, maximum_length, is_valid):
+    if is_valid:
+        validate_length_lte(value, maximum_length)
+    else:
+        with pytest.raises(ValidationError):
+            validate_length_lte(value, maximum_length)
+
+
+@pytest.mark.parametrize(
+    "value,minimum,is_valid",
+    (
+        (1, 2, False),
+        ('a','a', True),
```

Comment:
Rather than import Number, I just used `int`, is that cool? Everything seems to work fine.